src/platform/macos/device.rs

```diff
@@ -55,6 +55,12 @@ impl Device {
 			0
 		};
 
+		if let Some(l) = config.layer {
```

Comment:
I'd turn this into `if let Some(Layer::L2)`, what do you think?

Or at the very least:
```rust
if config.layer.filter(|l| l != Layer::L3).is_some() {
  return Err(ErrorKind::UnsupportedLayer.into()
}
```

Reply:
I'd go for the latter to be a bit extra defensive